packages/TelegramClient-UI.package/TCUChatsList.class/instance/chatItemClicked.event.from..st

```diff
@@ -1,13 +1,10 @@
 event handling
 chatItemClicked: aChatListItem event: anEvent from: aSender
 
-	self deselectChats.
-
 	(aChatListItem chatID ~= self selectedChat id)
-		ifTrue: [aChatListItem select.
+		ifTrue: [
 			self
-				selectedChat: aChatListItem chat;
-				triggerEvent: #newChatSelected with: aChatListItem chat]
-		ifFalse: [self
-			selectedChat: TCCNullChat new;
-			triggerEvent: #chatDeselected].
\ No newline at end of file
+				deselectChats;
+				selectChat: aChatListItem.]
+		ifFalse: [
+			self deselectChats.].
```

Comment:
just do `deselectChats` no matter what.

Reply:
Yeah we had funny behavior here: When we deselect chats before the condition, the `self selectedChat id` is the one from `TCCNullChat` and thus the condition is instanly true again, which leads to selecting the chat again that we wanted to deselect. What do you think of something like that:
```
| selectOtherChat |	

selectOtherChat := aChatListItem chatID ~= self selectedChat id.
self deselectChats.
selectOtherChat ifTrue: [ self selectChat: aChatListItem. ].
```

Reply:
if it works this would be a lot better